src/main/java/org/b612foundation/adam/astro/AstroUtils.java

```diff
@@ -7,17 +7,26 @@
 import java.time.temporal.ChronoUnit;
 import java.time.temporal.JulianFields;
 
-import static org.b612foundation.adam.astro.AstroConstants.DAY_TO_SEC;
+import static org.b612foundation.adam.astro.AstroConstants.*;
 
 public class AstroUtils {
 
     public static LocalDateTime localDateTimefromMJD(double mjd) {
         long dayComponent = (long)mjd;
-        double timeComponentSeconds = (mjd - dayComponent) * DAY_TO_SEC;
-        long timeSeconds = (long)timeComponentSeconds;
-        long timeNanoSeconds = (long)(timeComponentSeconds - timeSeconds);
-        LocalDate date = LocalDate.MIN.with(JulianFields.MODIFIED_JULIAN_DAY, dayComponent);
-        LocalTime time = LocalTime.MIDNIGHT.plusSeconds(timeSeconds).plusNanos(timeNanoSeconds);
+        double timeComponentDays = (mjd - dayComponent);
+        double hoursFull = timeComponentDays * DAY_TO_HR;
+        long hoursComponent = (long) hoursFull;
+        double minutesFull = (hoursFull - hoursComponent) * HR_TO_MIN;
+        long minutesComponent = (long)  minutesFull;
+        double secondsFull = (minutesFull - minutesComponent) * MIN_TO_SEC;
+        long secondsComponent = (long ) secondsFull;
+        long nanoComponent = (long)((secondsFull - secondsComponent) * SEC_TO_NANO);
+        LocalDate date = LocalDate.MIN.with(JulianFields.MODIFIED_JULIAN_DAY, dayComponent) ;
+        LocalTime time = LocalTime.MIDNIGHT
```

Comment:
Worth a comment why adding seconds in one shot as opposed to hours+minutes+seconds was a problem?